Get game to work

vincentvvv opened the pull request.

vincentvvv committed:
fix game

vincentvvv commented on index.html:
```diff
@@ -1,33 +1,34 @@
+      crossorigin="anonymous"
+    ></script>
+
+    <script src="scripts/game.js"></script>
```
for each script file you need to add the file here

vincentvvv commented on index.html:
```diff
@@ -1,33 +1,34 @@
+    <div id="start" class="box">
+      <button onclick="start()">Start</button>
+    </div>
+    <div id="draw" class="box"></div>
```
use `div` instead of `p`. Keep `p` only for text stuff

vincentvvv commented on scripts/enterGame.js:
```diff
@@ -1,7 +1,7 @@
-let title = document.getElementById("title").innerHTML;
-let start = document.getElementById("start").innerHTML;
-let draw = document.getElementById("draw").innerHTML;
+let titleElement = document.getElementById("title");
```
if you are referencing an element, add `element` to the suffix to make it more clear.
Also you want the element object, and not the `innerHTML`

vincentvvv commented on scripts/enterGame.js:
```diff
@@ -13,35 +13,33 @@ function start() {
-    drawElement.appendChild(drawNameBox);
-    drawElement.appendChild(drawSelection);
-    drawElement.appendChild(drawDisplay);
+    drawElement.appendChild(drawNameBox());
```
You are trying to append `drawNameBox` which is a function. I moved the login in here and made the functions return the HTML element

vincentvvv commented on scripts/enterGame.js:
```diff
@@ -13,35 +13,33 @@ function start() {
 // Setup the codes that would be written to draw the name box.
 function drawNameBox(){
     let nameBox = document.createElement("p");
     let input = `<input id="user" type="text" value="" placeholder="Enter Name" />`;
```
Let's stay consistent and continue to use `createElement` to create the elements

vincentvvv commented on scripts/game.js:
```diff
@@ -1,96 +1,91 @@
-        } else {
-            botOption = "Scissors";
-        }
+function drawUIGame() { 
```
you have `drawUI` in the other js file. There is a naming conflict, so I had to rename this one.

vincentvvv commented on scripts/game.js:
```diff
@@ -1,96 +1,91 @@
-            botOption = "Scissors";
-        }
+function drawUIGame() { 
+  const displayElement = document.getElementById("display");
```
since the `display` element doesn't exist when the page starts up, we need to find it everytime we want to draw the UI

deechow commented on scripts/enterGame.js:
```diff
@@ -13,35 +13,33 @@ function start() {
 
 // Draws the entire game, by writing in all the new codes.
-function drawUI {
+function drawUI() {
```
When I open it, there's this line here `const displayElement = document.getElementById("display");` which doesn't affect the codes at all. If I change it to "drawElement", it would delete the rock paper scissors buttons.

This code doesn't exists in your edit's comment, or the changes highlights but it appeared in the new codes (not the old ones). I will remove this line, but I'm also lowkey writing this to remember this exists, so I can recover it if it turns out I'll need it. (A little of a mental note.)

deechow replied:
Nevermind, I see where it came from (you've moved it from the game.script into here). IDK why it doesn't say it was added in line 27.

deechow committed:
Almost done

Everything seems fine except 1 (major) problem. When you hit the "Main Menu" button, and then hit "Start" again, everything but the title disappears.

deechow committed:
Finish

Removing the drawElement.remove lines fixes them... I do not know why but it works perfectly now...
